src/components/Forms/Subscribe.js

```diff
@@ -16,98 +16,140 @@ const SubscribeSchema = Yup.object().shape({
     .required('Required'),
 })
 
-const SignUp = ({ subscribed = false }) => (
-  <div>
-    <h2>Join the Newsletter</h2>
-    <Formik
-      initialValues={{
-        email_address: '',
-        first_name: '',
-      }}
-      validationSchema={SubscribeSchema}
-      onSubmit={values => {
-        // setTimeout(() => {
-        //   alert(JSON.stringify(values, null, 2))
-        // }, 500)
-        fetch('https://app.convertkit.com/forms/834199/subscriptions', {
+const PostSubmissionMessage = ({ response }) => {
+  console.log('Post submission response: ', response)
+
+  return <div>You have submitted with a status of {response.status}</div>
+}
+
+class SignUp extends React.Component {
+  state = {
+    submitted: false,
+  }
+
+  async handleSubmit(values) {
+    this.setState({ submitted: true, loading: true })
```

Comment:
set the state that the form has been submitted, and it is currently loading (renders the submitting message on line 68)